comictaggerlib/taggerwindow.py

```diff
@@ -2151,14 +2169,25 @@ def load_archive(self, comic_archive: ComicArchive) -> None:
 
         self.config[0].internal__last_opened_folder = os.path.abspath(os.path.split(comic_archive.path)[0])
         self.comic_archive = comic_archive
+
+        self.metadata, success = self.overlay_ca_read_style(self.load_data_styles, self.comic_archive)
+        if not success:
+            self.exception(f"Failed to load metadata for {self.comic_archive.path}, see log for details\n\n")
+
+        self.update_ui_for_archive()
+
+    def overlay_ca_read_style(self, load_data_styles: list[str], ca: ComicArchive) -> tuple[GenericMetadata, bool]:
+        md = GenericMetadata()
+        success = True
```

Comment:
The `success` from the `overlay_ca_read_style` is only required because of the exception window created by `load_archive` with an error loading the style. If the `success` return is removed, how would you recommend covering that?

Maybe this isn't the intention for `ComicArchive` but could `self.md` be `None` `dict[str, GenericMetadata | None]` if there is an error with a style?

Reply:
On line 1751 you have a `success` variable from `overlay_ca_read_style`. Why don't you just stop it there instead of processing a file that you know has an error.

On line 1208 it's loading it back in after it attempted to save if `success` is false that means more than likely that CT broke the archive, or that the archive was broken before, tell the user that an error happened.

Reply:
I was trying to replicate what was already happening with the minimal amount of changes. Maybe I am misunderstanding what is already there:

1751 change. Before it would try and except with an empty `md`, log the error and carry on with an empty GMD. The `overlay_ca_read_style` should be doing the same?

The only time that doesn't happen is the `self.exception` with `load_archive` and why the `success` return as mentioned. Am I missing something that was happening before or should the `logger` exceptions not be "quiet"? Or:

```python
    def overlay_ca_read_style(self, load_data_styles: list[str], ca: ComicArchive) -> tuple[GenericMetadata, dict | None]:
        md = GenericMetadata()
        error = None
        try:
            for style in load_data_styles:
                metadata = ca.read_metadata(style)
                md.overlay(metadata)
        except Exception as e:
            error = e

        return md, error
```
And then the thing calling `overlay_ca_read_style` logs/displays the message?

```python
md, error = self.overlay_ca_read_style(self.load_data_styles, self.comic_archive)
        if error:
            logger.error("Failed to load metadata for %s: %s", self.ca.path, error)
            # additional if required
            self.exception(f"Failed to load metadata for {self.ca.path}: {error}\n\n")
```
I was trying to leave small footsteps.

Reply:
> I was trying to replicate what was already happening with the minimal amount of changes

There's your problem. Don't try to keep everything exactly as it was. You created a new function make use of it the best that you can. If it needs to change then change the created function. If the calling function needs to change then change the calling function. With this project don't assume that the previous code was done the "correct way" it's just the code that happened to be there, maybe it's correct maybe it's not.

In a PR if you create a new function to consolidate code and you give it a "success" return value I would expect it to be used not, arbitrarily ignored.